### PR TITLE
Fix: Throw error after parsing protocol from proxied response

### PR DESCRIPTION
**Pull request by dpgraham**
* Before it was throwing immediately which meant we weren't parsing the protocol and weren't showing the proper error responses
* If proxied W3C error doesn't have an error, make it `Unknown`
* Also removed usage of `static` because it's not part of standards yet

### CODE REVIEW

File: test/protocol/protocol-specs.js [hunk 1]
```diff
@@ -31,7 +31,9 @@ describe('Protocol', async function () {
       parseProtocol({protocol: 'MJSONWP', value}).should.eql({isW3C: false, isMJSONWP: true, value});
     });
     it('should throw if {protocol: "MJSONWP", error}', function () {
-      (() => parseProtocol({protocol: 'W3C', error: new Error('some error')})).should.throw(/some error/);
+      //(() => ).should.throw(/some error/);
```
Comment (mykola-mokhnach) [inline test/protocol/protocol-specs.js]: do we need the comment?

File: lib/protocol/protocol.js [hunk 2]
```diff
@@ -273,6 +273,9 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
       if (parsedDriverRes) {
         isW3C = parsedDriverRes.isW3C;
         isMJSONWP = parsedDriverRes.isMJSONWP;
+        if (driverRes.error) {
```
Comment (mykola-mokhnach) [inline lib/protocol/protocol.js]: driverRes or parsedDriverRes?

File: lib/protocol/errors.js [hunk 3]
```diff
@@ -811,6 +811,10 @@ function getResponseForW3CError (err) {
 
   httpStatus = err.w3cStatus;
 
+  if (!error) {
```
Comment (mykola-mokhnach) [inline lib/protocol/errors.js]: it might be handy to mention that `error` here is the actual string signature

File: lib/basedriver/driver.js [hunk 4]
```diff
@@ -412,4 +398,18 @@ for (let [cmd, fn] of _.toPairs(commands)) {
   BaseDriver.prototype[cmd] = fn;
 }
 
+BaseDriver.DRIVER_PROTOCOL = {
```
Comment (mykola-mokhnach) [inline lib/basedriver/driver.js]: static look better AFAIK

File: lib/protocol/protocol.js (hunk 2 above)
Comment (dpgraham) [reply lib/protocol/protocol.js]: They would both have the same error, but I'll change it to `parsedDriverRes` because that's more obvious.

**Comment by dpgraham** [comment]
Fixes pushed

File: lib/protocol/errors.js [hunk 5]
```diff
@@ -804,16 +804,20 @@ function getResponseForW3CError (err) {
   if (isErrorType(err, errors.BadParametersError)) {
```
Comment (mykola-mokhnach) [inline lib/protocol/errors.js]: why this condition is needed?

Comment (dpgraham) [reply lib/protocol/errors.js]: JSONWP are almost all 500, except for `BadParameters`, `NotYetImplemented` and `NoSuchDriver`.

File: lib/protocol/protocol.js [hunk 6]
```diff
@@ -465,14 +468,15 @@ function parseProtocol (driverRes) {
     return null;
   }
 
+  let isW3C = driverRes.protocol === BaseDriver.DRIVER_PROTOCOL.W3C;
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/protocol/protocol.js]: the function does not throw anymore. Also the returned result might be different

File: lib/protocol/errors.js (hunk 5 above)
Comment (mykola-mokhnach) [reply lib/protocol/errors.js]: this might be handy as a comment

File: lib/protocol/errors.js [hunk 7]
```diff
@@ -804,16 +804,20 @@ function getResponseForW3CError (err) {
...
+  if (!w3cErrorString) {
+    w3cErrorString = UnknownError.error();
+  }
+
   let httpResBody = {
     value: {
-      error,
+      error: w3cErrorString,
       message: err.message,
       stacktrace: err.stack,
```
[the remark this answers is not in the document]
Comment (dpgraham) [reply lib/protocol/errors.js]: This gives us the stacktrace that was provided by the inner driver. So in the case of a Java inner driver being proxied, we would get the Java stack, which IMO is more helpful.

**Comment by dpgraham** [comment]
Published to 2.30.1

**Pull request merged**